Operators : 

- Arithmetic Operators: +, -, *, /, %, ++, --
- Comparison Operators: ==, !=, >, <, >=, <=
- Logical Operators: &&, ||, !
- Assignment operator 
- Indentity Operators 
- Bitwise ***********************************************
- Memebership Operaor

Operators       

+ addition 
- sub 
* mul 
/ divison 
// floor divison - The largest interger less than or equal to division result a//b
% modulus 
** power 



In [2]:
10//2

5

In [1]:
10/2

5.0

In [4]:
10//3

3

In [3]:
10/3

3.3333333333333335

# FoodExpress currently has fragmented data across order,payment,restaurents,delivery,customer,applications 

# The company needs a scabale,reliable,secure and automated data engineering platform to ingest batch and streaming data, validate it, store it  in data lake and warehouse and provide curated datasets for dashbords, analytics and machine learning use cases


# The platform should support historical load, incremntal load, monitoring, alearting , CI/CD deployments, data quality checks, failure handing and reprocessing and access control 

# Data Ingestion 
# Raw Layer ---> 
# Data validation 
# Transformation 
# Curated Layer --->
# Warehouse Load 
# Dashboard Tables 
# Monitoring 
# Alerting 
# CI/CD 
# Production Support 


CEO 
Revenue Today : 2 Crore 
Orders Today 
Active Custmers 
Delivery Sucess Rate 

Finance Team 
Marketing Team 
Operations Team 
Restaurant Team 
Customer Support Team 
Data Team

Data Science Team 





1. Arithmetic Operators

Used to perform mathematical calculations.

Operator	    Meaning	    Example	    Result
+	Addition	            10 + 5	    15
-	Subtraction	            10 - 5	    5
*	Multiplication	        10 * 5	    50
/	Division	            10 / 5	    2.0
//	Floor Division	        10 //3	    3
%	Modulus (remainder)	    10 % 3	    1
**	Exponent	            2 ** 3	    8

a = 10
b = 3

print(a + b)   # 13
print(a - b)   # 7
print(a * b)   # 30
print(a / b)   # 3.3333
print(a // b)  # 3
print(a % b)   # 1
print(a ** b)  # 1000

2. Comparison (Relational) Operators

Used to compare two values.

Returns either True or False

Operator	Meaning
==	        Equal to
!=	        Not equal to
>	        Greater than
<	        Less than
>=	        Greater than or equal
<=	        Less than or equal

a = 10
b = 20

print(a == b)   # False
print(a != b)   # True
print(a > b)    # False
print(a < b)    # True
print(a >= b)   # False
print(a <= b)   # True

3. Assignment Operators

Used to assign values to variables.

Operator	    Example	Equivalent To
=	x = 5	    Assign 5
+=	x += 3	    x = x + 3
-=	x -= 3	    x = x - 3
*=	x *= 3	    x = x * 3
/=	x /= 3	    x = x / 3
//=	x //= 3	    x = x // 3
%=	x %= 3	    x = x % 3
**=	x **= 3	    x = x ** 3

x = 10

x += 5
print(x)   # 15

x *= 2
print(x)   # 30

4. Logical Operators

Used to combine conditions.

Operator	    Meaning
and	            True if both conditions are True
or	            True if at least one condition is True
not	            Reverses the result

a = 10
b = 20

print(a > 5 and b > 10)   # True
print(a > 15 or b > 10)   # True
print(not(a > 5))         # False

Truth Table
AND
A	B	    A and B
T	T	    T
T	F	    F
F	T	    F
F	F	    F

OR
A	B	    A or B
T	T	    T
T	F	    T
F	T	    T
F	F	    F

5. Bitwise Operators

Operate on binary numbers (bits).

Operator    	Meaning
&	            AND
^	            XOR
~	            NOT
<<	            Left Shift
>>	            Right Shift

a = 5   # 0101
b = 3   # 0011

print(a & b)   # 1
print(a | b)   # 7
print(a ^ b)   # 6

6. Membership Operators

Used to test whether a value exists in a sequence.

Operator	    Meaning
in	            Present
not in	        Not Present

numbers = [10, 20, 30]

print(20 in numbers)       # True
print(40 in numbers)       # False
print(40 not in numbers)   # True

name = "Python"

print("Py" in name)      # True
print("Java" in name)    # False

7. Identity Operators

Used to compare object identity (memory location).

Operator	        Meaning
is	                Same object
is not	            Different object

a = [1, 2, 3]
b = a
c = [1, 2, 3]

print(a is b)      # True
print(a is c)      # False

print(a == c)      # True

Difference Between == and is

a = [1,2]
b = [1,2]

print(a == b)  # True (values same)
print(a is b)  # False (different objects)

In [ ]:
== equal to : Check weather two values are equal or not 
!=  Not equals 
> Greater than 
< Less than 
>= Greater than or equal to 
<= Less than equal to 

In [2]:
a=20
b=10 
print(a==b)

False


In [3]:
10 != 10 

False

In [ ]:
a=10 
a==10 

In [5]:
sal=40000
print(sal>50000)

False


In [7]:
print(20<10)

False


In [9]:
print(20>=10)

True


In [ ]:
#Logical Operators: &&, ||, !

# c,java,javascript 
# && AND 
# || OR 
# ! NOT 

# a&&B Invalid in python 
# a || b Invalid in python 
# !a Invalid 

# In python 

# and
# or 
# not 

# a and b 
# a or b 
# not a 


# And : Returns True only if ALL conditions are True 

condition 1 and condition 2  

# print(True and True ) - True 
# print(False and True ) - False 

# A       b       A and B 

# True    True    True
# True    False   False
# False   True    False 
# False   False   False 



In [10]:
# # Assingment Operators 

# =   Assign 

# +=  Add & Assign 

# -=  Substract Assign 

# *= Multiply Asssign 

# /= Divide Assign 

# //= Floor divide assign 

# %=  Modulus Assign 

# a,b,c=10,20,30

x=10 
x=x+5

In [11]:
x

15

In [ ]:
x=10
x +=10 # x+10 =20 : x=20

In [13]:
x

20

In [ ]:
# Indentity Operators 

# is  same object in memory 

# is not  Difrrent Object in memory - Do they pointing to diffrent objects? 



In [15]:
a=10
b=10

print(a is not b)

False


In [ ]:
# Memebership Operaor

# in  - Present in collection 

# not in - Not present in collection 


# syntax:

# value in collection

# value not in collection

# fruits=["apple","banana"]

# if "mango" in fruits:
#     print("found")



In [19]:
list1=[10,20,30]

list2=[10,20,30]

print(list1==list2)

print(list1 is list2)

True
False
